c7n/resources/rds.py

```diff
@@ -832,6 +832,51 @@ def set_retention_window(self, resource, retention, copy_tags):
             CopyTagsToSnapshot=copy_tags)
 
 
+@actions.register('set-public-access')
+class RDSSetPublicAvailability(BaseAction):
+    """
+    This action allows for toggling an RDS instance
+    'PubliclyAccessible' flag to true or false
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: enable-rds-public-accessibility
+                resource: rds
+                filters:
+                  - PubliclyAccessible: false
+                actions:
+                  - type: set-public-access
+                    state: true
+    """
+
+    schema = type_schema(
+        "set-public-access",
+        state={'type': 'boolean'})
+    permissions = ('rds:ModifyDBInstanbce',)
+
+    def set_accessibility(self, r):
+        client = local_session(self.manager.session_factory).client('rds')
+        client.modify_db_instance(
+            DBInstanceIdentifier=r['DBInstanceIdentifier'],
+            PubliclyAccessible=self.data.get('state', False))
+
+    def process(self, rds):
+        with self.executor_factory(max_workers=2) as w:
+            futures = []
+            for r in rds:
+                futures.append(w.submit(
+                    self.set_accessibility, r))
+            for f in as_completed(futures):
+                if f.exception():
+                    self.log.error(
```

Comment:
can you log the db id here, basically switch futures to a dictionary to associate the future to the db, and then derereference and log. else the log isn't really helpful in identifying an issue.